finality-aleph/src/network/gossip/service.rs

```diff
@@ -77,6 +78,49 @@ impl Display for Error {
     }
 }
 
+#[derive(Debug)]
+pub enum GossipServiceError {
+    NetworkStreamTerminated,
+    AuthorizationStreamTerminated,
+    BlockSyncStreamTerminated,
+    UnableToForwardMessageToUser,
+}
+
+impl fmt::Display for GossipServiceError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            GossipServiceError::NetworkStreamTerminated => write!(f, "Network event stream ended."),
+            GossipServiceError::AuthorizationStreamTerminated => {
+                write!(f, "Authentication user message stream ended.")
+            }
+            GossipServiceError::BlockSyncStreamTerminated => {
+                write!(f, "Block sync user message stream ended.")
+            }
+            GossipServiceError::UnableToForwardMessageToUser => {
+                write!(f, "Cannot forward messages to user.")
+            }
+        }
+    }
+}
+
+impl GossipServiceError {
```

Comment:
Same.

Reply:
done

Reply:
This and the other I left unresolved were not renamed to just `Error` – is there a reason for that?

Reply:
`ServiceInterface` uses `Error`.